src/test/java/org/generationcp/middleware/api/germplasm/pedigree/GermplasmPedigreeServiceImplIntegrationTest.java

```diff
@@ -0,0 +1,329 @@
+package org.generationcp.middleware.api.germplasm.pedigree;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.generationcp.middleware.IntegrationTestBase;
+import org.generationcp.middleware.domain.germplasm.GermplasmDto;
+import org.generationcp.middleware.manager.DaoFactory;
+import org.generationcp.middleware.pojos.Germplasm;
+import org.generationcp.middleware.pojos.Method;
+import org.generationcp.middleware.pojos.MethodType;
+import org.generationcp.middleware.pojos.Name;
+import org.generationcp.middleware.pojos.Progenitor;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+public class GermplasmPedigreeServiceImplIntegrationTest extends IntegrationTestBase {
+
+	private DaoFactory daoFactory;
+
+	private Method derivativeMethod;
+
+	private Method generativeMethod;
+
+	private Method maintenanceMethod;
+
+	private Integer userId;
+
+	private Integer DRVNM_ID;
+
+	@Autowired
+	private GermplasmPedigreeService germplasmPedigreeService;
+
+	@Before
+	public void setUp() {
+		this.daoFactory = new DaoFactory(this.sessionProvder);
+		this.derivativeMethod = this.createBreedingMethod(MethodType.DERIVATIVE.getCode(), -1);
+		this.generativeMethod = this.createBreedingMethod(MethodType.GENERATIVE.getCode(), 2);
+		this.maintenanceMethod = this.createBreedingMethod(MethodType.MAINTENANCE.getCode(), -1);
+		this.userId = this.findAdminUser();
+		this.DRVNM_ID = this.daoFactory.getUserDefinedFieldDAO().getByTableTypeAndCode("NAMES", "NAME", "DRVNM").getFldno();
+	}
+
+	@Test
+	public void testGetGermplasmPedigreeTree_When_DerivativeAndIncludeDerivativeLinesIsFalse() {
```

Comment:
I think we can simplify as testXX_WhenYYY (no need to for "when" to be surrounded by "_")

Reply:
Done